Animations - Only in Offline mode

harshpurwar opened the pull request:
This PR adds support for animated lines and comets.

Sample code to try:
```
t = 0:pi/200:2*pi;
x = cos(3*t);
y = sin(2*t).*cos(3*t);
comet(x,y);
pf = fig2plotly(gcf, 'offline', 1, 'treatAs', 'comet')
```

**Note: This is supported in offline mode.**


![1](https://user-images.githubusercontent.com/38316470/134906093-a6ab7768-bdd1-4c12-9bc8-791a3fb3a54f.gif)
![2](https://user-images.githubusercontent.com/38316470/134906097-90b93a1e-f1bc-43ce-b95a-66aa75371538.gif)
![3](https://user-images.githubusercontent.com/38316470/134906099-be7c8ae3-a609-4a59-9c06-1452a2f1937d.gif)
![5](https://user-images.githubusercontent.com/38316470/134906102-6f176f30-8779-4fa8-be41-57fe8dd75668.gif)

gilbertogalvis commented:
Works fine for me!

gilbertogalvis reviewed (approved):
Excellent work! My only suggestion (coming from @jackparmer) is the request to use **camelCase** for code development.

harshpurwar commented:
Added options to specify FrameDuration and FrameTransitionDuration to control the speed of the animations.
Added support for comet3 and 3d animated line.

gilbertogalvis reviewed (commented):
When run updated code, it turns out following error

<img width="531" alt="Screen Shot 2021-09-29 at 11 23 31 AM" src="https://user-images.githubusercontent.com/56391490/135299661-aac74079-9b09-4788-a3aa-c31b5e8c793c.png">

Reviewing the code I found an error between lines #267 and #268 in `plotlyfig.m` file (please fix it. It is a missing `end` command).

Anyway when I fix previous error and run the following code

```
[xmat,ymat,zmat] = peaks(100); 
xvec = xmat(:);
yvec = ymat(:);
zvec = zmat(:);
comet3(xvec,yvec,zvec)

fig2plotly(gcf, 'offline', true, 'TreatAs', 'comet3');
```

This takes a long time to run and in at end throws a memory overflow error
